### PR TITLE
Only reject promise if there's a network issue

### LINKED ISSUE #27: Promises get rejected on any error, rather than network errors only
https://github.com/zooniverse/json-api-client/blob/df87d8a3ed1afa007aa2a9a4246901111f5adae4/src/make-http-request.coffee#L39-L40

We shouldn't be rejecting promises just because there's an error; that should be reserved for instances where we're unable to get a response from the server at all. This is (I suspect) the cause of zooniverse/panoptes-javascript-client#25

cc @eatyourgreens

### PR DESCRIPTION
**Pull request by rogerhutchings**
Will need to test for regression errors... or make this a major version bump, since the API is changing.

Fixes #27

### CODE REVIEW

**Comment by brian-c** [comment]
I don't really agree that this is the right behavior. The library is for working with resources, not making HTTP requests (this behavior would make sense for superagent though, for example). `resolve`-ing after an action should indicate that it was completed as intended, `reject`ing should indicate that it did not.

This would also require rewriting a good chunk of the main app.

I think the right change would be to include the status code in the error object. Or maybe a subclass of `Error` would be more appropriate? This one looks nice: https://www.npmjs.com/package/standard-http-error

I think this would only require rewriting the `handleError` method in panoptes-client's JSONAPIClient instance.

File: src/make-http-request.coffee [hunk 1]
```diff
@@ -44,12 +44,13 @@ makeHTTPRequest = (method, url, data, headers = {}, modify) ->
 
     req.end (error, response) ->
       delete getsInProgress[requestID]
+
       if error?.status is 408
         resolve makeHTTPRequest.apply null, originalArguments
-      else if error?
+      else if error? and response in error is false
```
Comment (brian-c) [inline src/make-http-request.coffee]: `in` here checks that an array includes an item. You probably want `of` (or `response not of error` in this case, [or `'response'` if you're checking the key, but I'm not really sure of the context]). Dumb CoffeeScript quirk.

**Comment by eatyourgreens** [comment]
Sounds good to me http://jsonapi.org/examples/#error-objects includes the status code in the first example. 

:+1: to re-writing panoptes-client's `handleError` method, eg. so that we can catch `300` redirects and `400` client-side errors on `resource.save()` so that we don't continuously retry bad requests from PFE.

**Comment by brian-c** [comment]
We're using an old prerelease version of JSON API, and I don't think error responses had been standardized at that point so we're really just wingin' it.

I used to have a Wayback Machine link to near the version we used, I'll see if I can find it and add it to the README.

**Comment by brian-c** [comment]
Replaced by #29.

**Pull request closed without merging**